### PR TITLE
[YogaKit] Improved the objective-c and swift api

### PR DESCRIPTION
**Pull request by hartbit**
Compared to what was planned, I added the `overflow` value which seemed missing. I had to modify the implementation a bit for all values which are backed by a `YGValue`, but we should probably enable the pixel dimensions in Objective-C and Swift somehow later.

### CODE REVIEW

File: YogaKit/YGLayout.m [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+YG_STYLE_EDGE_PROPERTY_UNIT_IMPL(lowercased_name, capitalized_name, YGEdgeBottom, Bottom)         \
+YG_STYLE_EDGE_PROPERTY_UNIT_IMPL(lowercased_name, capitalized_name, YGEdgeStart, Start)           \
+YG_STYLE_EDGE_PROPERTY_UNIT_IMPL(lowercased_name, capitalized_name, YGEdgeEnd, End)               \
+YG_STYLE_EDGE_PROPERTY_UNIT_IMPL(lowercased_name, capitalized_name, YGEdgeHorizontal, Horizontal) \
+YG_STYLE_EDGE_PROPERTY_UNIT_IMPL(lowercased_name, capitalized_name, YGEdgeVertical, Vertical)     \
+YG_STYLE_EDGE_PROPERTY_UNIT_IMPL(lowercased_name, capitalized_name, YGEdgeAll, )
+
+@interface YGLayout ()
+
+@property (nonatomic, weak, readonly) UIView* view;
```
Comment (emilsjolander) [inline YogaKit/YGLayout.m]: UIView *view

File: yoga/YGEnums.h [hunk 2]
```diff
@@ -14,28 +14,54 @@
 YG_EXTERN_C_BEGIN
 
 #define YGFlexDirectionCount 4
+#ifndef NS_ENUM
```
Comment (emilsjolander) [inline yoga/YGEnums.h]: Could we make this generate something short (less duplicated) ? Maybe by creating our own YG_ENUM macro which expands to one of the two alternatives.

File: YogaKit/YGLayout.m [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "YGLayout+Private.h"
+#import "UIView+Yoga.h"
+#import <yoga/Yoga.h>
+#import <UIKit/UIKit.h>
+
+extern YGValue YGValueMake(CGFloat value, YGUnit unit) {
```
Comment (emilsjolander) [inline YogaKit/YGLayout.m]: This function seems unused. Could you remove it?

File: YogaKit/YGLayout.h [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <UIKit/UIKit.h>
+#import <yoga/YGEnums.h>
+#import <yoga/YGValue.h>
```
Comment (emilsjolander) [inline YogaKit/YGLayout.h]: This import seems unused, could you remove it?

File: yoga/YGValue.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/**
```
Comment (emilsjolander) [inline yoga/YGValue.h]: Pulling this out into a separate file seems to not be needed for this PR. Could you revert this change?

**Comment by hartbit** [comment]
I implemented all your suggestions 😄  Thanks 👍

File: YogaKit/UIView+Yoga.h [hunk 6]
```diff
@@ -8,76 +8,10 @@
...
-/**
- Return a BOOL indiciating whether or not we this node contains any subviews that are included in Yoga's layout.
- */
-- (BOOL)yg_isLeaf;
-
-/**
- Mark that a view's layout needs to be recalculated. Only works for leaf views.
- */
-- (void)yg_markDirty;
+@property (nonatomic, readonly, retain) YGLayout *yoga;
```
Comment (dshahidehpour) [inline YogaKit/UIView+Yoga.h]: We should change them to `strong`

http://nshipster.com/associated-objects/

File: YogaKit/YGLayout.m [hunk 7]
```diff
@@ -0,0 +1,380 @@
...
+@synthesize isIncludedInLayout=_isIncludedInLayout;
+
++ (void)initialize
+{
+  YGSetExperimentalFeatureEnabled(YGExperimentalFeatureWebFlexBasis, true);
+}
+
+- (instancetype)initWithView:(UIView*)view
+{
+  if ([super init]) {
```
Comment (dshahidehpour) [inline YogaKit/YGLayout.m]: Nit, we should be assigning this to `self`, i.e.

`self = [super init]`

**Comment by emilsjolander** [comment]
@hartbit Don't worry about it. I'll fix this while mergin

**Comment by dshahidehpour** [comment]
Awesome job @hartbit!! I'm so excited to land this!

**Comment by ericvicenti** [comment]
It looks like the import of this commit into react-native causes CI tests to fail. Can somebody help fix this ASAP?

https://travis-ci.org/facebook/react-native/builds/190054212

**Comment by emilsjolander** [comment]
@ericvicenti looking now

**Comment by emilsjolander** [comment]
@ericvicenti Found issue. Pushing fix within next few hours

**Comment by hartbit** [comment]
@emilsjolander What was the problem?

**Comment by emilsjolander** [comment]
https://github.com/facebook/yoga/commit/0acb6201594739d069d8973ee8391dea9bfaf9c7

**Comment by hartbit** [comment]
Ah, of course!
